doc/cabal-package.rst

```diff
@@ -1454,24 +1435,25 @@ system-dependent values for these fields.
 
 .. pkg-field:: build-depends: library list
 
-    Declares the *library* dependencies required to build the current
-    package component; see :pkg-field:`build-tool-depends` for
-    declaring build-time *tool* dependencies. External library
-    dependencies should be annotated with a version constraint.
+    Declares the *library* dependencies (from the same package or other
+    packages) required to build the current package component. See
+    :pkg-field:`build-tool-depends` for declaring build-time *tool*
+    dependencies. Dependencies on libraries from another package should be
+    annotated with a version constraint.
 
     **Library Names**
 
-    External libraries are identified by the package's name they're
-    provided by, optionally followed by a colon and the library name
-    (available from ``cabal-version: 3.0``).
-    If the library name is absent, the main (unnamed) library will be used.
-    To refer to the main (unnamed) library explicitly, use the name of the
-    package (``foo:foo``).
-    Multiple libraries from the same package can be specified with the shorthand
-    syntax ``pkg:{lib1,lib2}```.
-
-    See section on :ref:`Multiple Public Libraries <publicsublibs>` for examples and more
-    information.
+    A library is identified by the name of the package that provides it,
+    optionally followed by a colon and the library's name (available from
+    ``cabal-version: 3.0``). If a library name is omitted, the package's 'main'
+    unnamed library will be used. To refer expressly to a package's 'main'
+    unnamed library, use the name of the package as the library name (for
+    example, ``my-package:my-package``). More than one library from the same
+    package can be specified with the shorthand syntax ``pkg:{lib1,lib2}``.
+
+    See the section on :pkg-section:`library` for more information about how
+    a package can specify library components for use by other components of the
+    same package or by other packages.
```

Comment:
```suggestion
    See :pkg-section:`library` for more information about declaring
    library components.
```
Be explicit about what this is about: declaring library components.

I think some of this is unnecessary and can be cut

Reply:
I have now shortened this. I preferred 'to specify' over 'to declare', because the former seems more frequent in the existing documentation. I don't think much turns on that (although editorial consistency would be nice to have).